### PR TITLE
[DD-408] Use PrismJS for code highlighting 

### PR DESCRIPTION
**Pull request by rsylvian**
Ticket: [DD-408](https://circleci.atlassian.net/browse/DD-408)
Preview: [link](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/rs/feat/add-prismjs-preview/?force-all
)

# Description
- Replace HighlightJS with PrismJS
- Update all `.adoc` & `.md` to use `shell` instead of `sh` (more [info](https://github.com/circleci/circleci-docs/pull/6326#discussion_r799119956))
- Add babel prismjs plugin to be able to only bring JS/CSS we need (rather than all PrismJS+plugins)
- Update [snippetFeedback experiment](https://github.com/circleci/circleci-docs/pull/6326/files#diff-fb79bc521a0dfe6d7459e382d8772b5809578ed1e0358d1cc42efb737901a7e0) to be compatible with PrismJS
-  Update highlight theme to closely match our in-ap config editor

# Reasons
Recently, we have been experiencing a few issues with highlightJS. We experimented with PrismJS and it is great for our use cases. 

# Screenshots

before:
screely-1644029702129

after:
screely-1643942689300

### CODE REVIEW

File: jekyll/_api/v1-reference.md [hunk 1]
```diff
@@ -179,7 +179,7 @@ The branch name should be url-encoded.
 
 You can download an individual artifact file via the API with an API-token authenticated HTTP request.
 
-```sh
+```shell
```
Comment (rsylvian) [inline jekyll/_api/v1-reference.md:182]: `sh` is not a supported language code for PrismJS. Have to change all of them to `shell`

**Review by chantelle11 — COMMENTED** [review]
Havn't got to do a big dive yet but noticed that when you make screen sizes smaller the numbering is wrong. [Link from preview](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/rs/feat/add-prismjs-preview/2.0/getting-started/?force-all)

Screen Shot 2022-02-03 at 10 15 06 PM

**Review by chantelle11 — COMMENTED** [review]
I can't seem to change to Japanese.

**Comment by rsylvian** [comment]
> Havn't got to do a big dive yet but noticed that when you make screen sizes smaller the numbering is wrong. [Link from preview](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/rs/feat/add-prismjs-preview/2.0/getting-started/?force-all)
> 
> Screen Shot 2022-02-03 at 10 15 06 PM

Good catch! It’s actually not wrong (there are 30 lines) but the wrapping is messing with whole thing. Will fix!

**Comment by rsylvian** [comment]
> I can't seem to change to Japanese.

Unfortunately, It never worked in the previews…

**Comment by chantelle11** [comment]
When I go to this PR [link](https://github.com/circleci/circleci-docs/pull/6306) The preview lets me go to JA [link](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/update-icons-preview/ja/)

File: jekyll/_config.yml [hunk 2]
```diff
@@ -74,7 +74,6 @@ liquid:
 
 asciidoc:
   attributes:
-      source-highlighter: rouge
```
Comment (rsylvian) [inline jekyll/_config.yml:77]: @rosieyohannan I need to remove this to make PrismJS work in `.adoc` files but I am wondering if there is some sort of impact on the generated pdf?

I tried to generate pdf locally with `./scripts/build_pdfs_asciidoc.sh` and I am still seeing highlighted code: 
Screen Shot 2022-02-04 at 6 05 39 PM

Am I in the clear?

**Comment by nramabad** [comment]
is the copy code button still a WIP? if not and it's a preview issue, can you add something to the forceall script that will fix it so we can see what it should look like?

**Comment by rsylvian** [comment]
> is the copy code button still a WIP? if not and it's a preview issue, can you add something to the forceall script that will fix it so we can see what it should look like?

Yup this is already in the PR. Make sure to have "?force-all" in the url when you open the preview and it will work

**Comment by nramabad** [comment]
> > is the copy code button still a WIP? if not and it's a preview issue, can you add something to the forceall script that will fix it so we can see what it should look like?
> 
> Yup this is already in the PR. Make sure to have "?force-all" in the url when you open the preview and it will work

oh thanks I see it now. but it doesn't seem to have a mousedown over button darkening effect? right now its hover (code) => hover (button) => click (button)

Is this intentional?

**Review by teesloane — COMMENTED** [review]
Nice work so far :)

Heads up - It looks like line numbers might be broken just on the guided tour pages:

image

**Comment by rsylvian** [comment]
> Nice work! The new theme looks really good. :)
> 
> Heads up - It looks like line numbers might be broken just on the guided tour pages:
> 
> image

Oh wow, that's wild! Good catch thank you!

**Comment by rsylvian** [comment]
> > > is the copy code button still a WIP? if not and it's a preview issue, can you add something to the forceall script that will fix it so we can see what it should look like?
> > 
> > 
> > Yup this is already in the PR. Make sure to have "?force-all" in the url when you open the preview and it will work
> 
> oh thanks I see it now. but it doesn't seem to have a mousedown over button darkening effect? right now its hover (code) => hover (button) => click (button)
> 
> Is this intentional?

Good catch! I'll add it!

File: jekyll/_config.yml (hunk 2 above)
Comment (rosieyohannan) [reply jekyll/_config.yml:77]: Yes that looks fine! I guess we didn't need that attribute :-D

**Review by chantelle11 — COMMENTED** [review]
`migrating-from-azuredevops` has a copy code button on main [site](https://circleci.com/docs/2.0/migrating-from-azuredevops/) but not [preview](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/rs/feat/add-prismjs-preview/2.0/migrating-from-azuredevops/?force-all)

Specifically GitHub Enterprise and force all is on
Screen Shot 2022-02-07 at 10 17 22 PM

**Review by chantelle11 — COMMENTED** [review]
Tool tip not working for me in preview `Copy to clipboard`
Screen Shot 2022-02-07 at 10 19 53 PM
Screen Shot 2022-02-07 at 10 20 07 PM

**Comment by rsylvian** [comment]
> `migrating-from-azuredevops` has a copy code button on main [site](https://circleci.com/docs/2.0/migrating-from-azuredevops/) but not [preview](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/rs/feat/add-prismjs-preview/2.0/migrating-from-azuredevops/?force-all)
> 
> Specifically GitHub Enterprise and force all is on Screen Shot 2022-02-07 at 10 17 22 PM
> 
> Screen Shot 2022-02-07 at 10 17 59 PM

Thanks, @chantelle11! This is a content issue and it's hard to catch all of them. This is fixed in https://github.com/circleci/circleci-docs/pull/6326/commits/a85ad268d16a77565891e7d3a5353797118e520c

**Comment by rsylvian** [comment]
> Tool tip not working for me in preview `Copy to clipboard` Screen Shot 2022-02-07 at 10 19 53 PM Screen Shot 2022-02-07 at 10 20 07 PM

Unfortunately, it is not a feature offered by PrismJS. I think we will have to do without it for now :)

File: jekyll/_cci2/browser-testing.md [hunk 3]
```diff
@@ -37,7 +37,7 @@ WebDriver can operate in two modes: local or remote. When run locally, your test
 
 If Selenium is not included in your primary docker image, install and run Selenium as shown below::
 
-```yml
+```yaml
```
Comment (tshen815) [inline jekyll/_cci2/browser-testing.md:40]: is this change also because `yml` is not a supported language code for PrismJS?

Comment (rsylvian) [reply jekyll/_cci2/browser-testing.md:40]: both `yml` & `yaml` keys are supported but just wanted to make everything consistent: https://prismjs.com/#supported-languages

File: jekyll/_cci2/runner-api.adoc [hunk 4]
```diff
@@ -293,8 +293,13 @@ Get the number of running tasks for a given resource class.
 
 ==== Examples
 
+<<<<<<< HEAD
+```shell
+curl -X GET https://runner.circleci.com/api/v2/runner/tasks/running?resource-class=test-namespace/test-resource \
+=======
 ```sh
 curl -X GET https://runner.circleci.com/api/v2/tasks/running?resource-class=test-namespace/test-resource \
+>>>>>>> 7626e45fc82df4308a3969ce8eb452b1c6422f5d
```
Comment (teesloane) [inline jekyll/_cci2/runner-api.adoc:302]: Looks like a merge conflict snuck in :)

File: jekyll/_cci2_ja/runner-api.adoc [hunk 5]
```diff
@@ -293,8 +293,13 @@ Get the number of running tasks for a given resource class.
 
 ==== サンプル
 
+<<<<<<< HEAD
+```shell
+curl -X GET https://runner.circleci.com/api/v2/runner/tasks/running?resource-class=test-namespace/test-resource \
+=======
 ```sh
 curl -X GET https://runner.circleci.com/api/v2/tasks/running?resource-class=test-namespace/test-resource \
+>>>>>>> 7626e45fc82df4308a3969ce8eb452b1c6422f5d
```
Comment (teesloane) [inline jekyll/_cci2_ja/runner-api.adoc:302]: Another sneaky merge conflict :)

**Review by teesloane — CHANGES_REQUESTED** [review]

File: jekyll/_cci2_ja/runner-api.adoc (hunk 5 above)
Comment (rsylvian) [reply jekyll/_cci2_ja/runner-api.adoc:302]: 🤭

Comment (rsylvian) [reply jekyll/_cci2_ja/runner-api.adoc]: @teesloane done!

**Comment by tshen815** [comment]
This looks great to me! I couldnt find a snippet that was missing the numbering and format. Snippets look great in different views as well along with mobile. 🎉 🎉

**Review by tshen815 — APPROVED** [review]
This looks great to me! I couldnt find a snippet that was missing the numbering and format. Snippets look great in different views as well along with mobile. 🎉 🎉

**Comment by rsylvian** [comment]
@devinmarieb Yup! The preview is several versions behind because something is breaking in our CI workflow. Trying to fix it currently. It should be good once everything builds again but I'll reach out to you for one last check if you want 🙌

**Comment by devinmarieb** [comment]
@rsylvian No worries I don't need to check, I just wanted to verify 👍

**Review by Mayuko-Sa — APPROVED** [review]

**Review by teesloane — APPROVED** [review]

**Pull request merged**